snow/networking/handler/handler.go

```diff
@@ -742,8 +746,16 @@ func (h *handler) handleSyncMsg(ctx context.Context, msg Message) error {
 			zap.String("messageOp", op),
 			zap.Stringer("message", body),
 		)
-		return nil
-	// Connection messages can be sent to the currently executing engine
+		s, ok := engine.(simplexEngine)
```

Comment:
That's one way of doing it. Another way would be to have a default implementation in the other engines that would return an error.

Why is this way superior? I don't have a strong opinion, just wondering what are your thoughts about this.